kubeshop/testkube-dashboard
# tech: add universal permissions mechanism

## Changes

- Add permissions mechanism, same as in [**Cloud version**](https://github.com/kubeshop/testkube-cloud-ui/pull/203/files)
- Adjust some files to use the same code as in Cloud
   - Add option to disable `CreatableMultiSelect`
   - Add option to modify `enabled`/`isEditable` state of `ConfigurationCard`
   - Adjust `EmptyListContent` to display a different message when it can't add/run a new one

## Fixes

- OSS part of https://github.com/kubeshop/testkube/issues/3693

## How to test it

- Build app, check if everything works, change `has` method in `BasePermissionsResolver` for tests with specific permissions rule

## screenshots

-

## Checklist
